### PR TITLE
feat: add `autoComplete` to most inputs / fields.

### PR DESCRIPTION
**Pull request by montezume**
#### Summary

Adds prop `autoComplete` to fields and inputs.

### CODE REVIEW

**Comment by montezume** [comment]
@tdeekens but modern browsers accept a whole whack of values for autocomplete...
https://stackoverflow.com/questions/34712926/how-browsers-store-data-for-autocomplete-and-where/34776932#34776932

File: src/components/fields/select-field/select-field.form.story.js [hunk 1]
```diff
@@ -96,7 +96,8 @@ storiesOf('Examples|Forms/Fields', module)
                 errors={formik.errors.animal}
                 isRequired={true}
                 touched={formik.touched.animal}
-                name="animal"
+                name="name"
+                isSearchable={true}
```
Comment (jonnybel) [inline src/components/fields/select-field/select-field.form.story.js]: Is this supposed to be here?

**Comment by jonnybel** [comment]
This is the type of prop that I'd personally prefer to spread directly to the HTML element in a `remainingProps` fashion, because we're basically just adding more code, tests and documentation for stuff that is already native to the HTML, and seems very unnecessary. (Other examples are props like isDisabled, readOnly, autoFocus, etc, as well as mouse events and whatnot)

But I know this would require a complete refactor of all inputs/fields (and even other components in our UI-kit), so I just wanted to state this for consideration.

That being said, nothing wrong with this PR in particular considering how we're doing things so far 😄

File: src/components/fields/select-field/select-field.form.story.js (hunk 1 above)
Comment (montezume) [reply src/components/fields/select-field/select-field.form.story.js]: Nice catch

**Comment by montezume** [comment]
@jonnybel really good point. It just adds noise to the component and especially to the readme the way it is now...

**Comment by montezume** [comment]
@jonnybel do you want to open an issue and list the props that should be whitelisted for this spreading behaviour? I guess aria- props, etc...

**Comment by jonnybel** [comment]
Yes, I can do that.

**Pull request merged**